proposed/2022/DotnetUpdatePackage.md

```diff
@@ -0,0 +1,179 @@
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+Staying current on the latest package version is a major challenge for modern .NET developers. Developers would like an update all experience in the .NET CLI similar to what they can do in modern IDEs like Visual Studio and checking all packages to then be updated in bulk. This feature will support the bulk updating of eligible `<PackageReference>` elements throughout their project files and solutions.
+
+`dotnet update package` can automatically update any outdated package(s) to the latest version depending on the update strategy provided by the user.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+To rid developers of "DLL Hell" in managing modern .NET dependencies, we would like to implement this feature to rid developers of daily pain of managing their dependencies and staying current with the latest versions of packages to maintain a secure supply chain and new package feature functionality.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+This command will update dependencies to the latest version respecting the semver constraints of your package and its transitive dependencies. It will also install any missing packages if an newer version brings them in. It will take into account lock files and central package management as locations that exist in addition to common definitions of `<PackageReference>` in project file(s).
+
+To ensure compatibility with existing package definitions in various files, we should not overwrite or simplify XML elements, but rather only update package version attributes.
+
+![](../../meta/resources/dotnetupdatepackage/dotnetupdatepackage.png)
+
+```
+dotnet update [<PROJECT>|<SOLUTION>] package <PACKAGE_NAME>
+    [-f|--framework <FRAMEWORK>] [--interactive]
+    [-n|--no-restore] [--package-directory <PACKAGE_DIRECTORY>]
+    [--prerelease] [-s|--source <SOURCE>] [-v|--version <VERSION>]
+    [--dry-run] [--highest-minor] [--highest-patch]
+
+dotnet update package -h|--help
+```
+
+#### Arguments
+
+- PROJECT | SOLUTION
+
+The project or solution file to operate on. If not specified, the command searches the current directory for one. If more than one solution or project is found, an error is thrown.
+
+- PACKAGE_NAME
+
+The package reference to add. Package name wildcards should be supported and update a subset of a glob-style pattern package name. i.e. `Microsoft.*` would update Microsoft packages. The commonly used symbols to support are:
+
+- `*` - zero or more characters.
+- `?` - single occurrence of any character.
+- `.` - literal "." character
+
+#### Options
```

Comment:
Any plan to include a `--compatible` or have it automatically update packages to the latest compatible version if the package latest stable version doesn't support the TFM?

It's referenced in this comment: https://github.com/NuGet/Home/issues/4103#issuecomment-802598457

It's probably beyond the scope an initial release, but a maybe a cool idea 🤷‍♂️

Reply:
Not yet. We don't have compat client tooling yet outside of one field. Once we can use the server side stuff, perhaps we can light up this.